PublicMapping/districtbuilder
# Project template reference layers

## Overview

Adds an option to associate a project template with a reference layer, and then copies that that reference layer to the project when creating a project from that template.

Part of this includes a larger refactor to how creating templates from projects works (https://github.com/PublicMapping/districtbuilder/commit/85618ae611be4c9625279b88d9eabf39506ee631), moving most of the logic for copying fields from template -> project from the client to the server.

### Checklist


### Demo

![ref-layers](https://user-images.githubusercontent.com/4432106/134918615-6b9958dd-ef8d-4dff-8178-1bde3e99927f.gif)


### Notes

I considered instead of copying reference layers, changing the ManyToOne relation to a ManyToMany and 

## Testing Instructions

 - `scripts/update`
- Add a reference layer to an existing project, then update it to instead point to a project template:
  `UPDATE reference_layer SET project_id = NULL, project_template_id = '<uuid_here>';`
- Create a project using the project template you just set your reference layer to point to
  - The newly created project should start with that reference layer by default
- There should be no regressions in creating a project from a template on either the organization or create project screens

Closes #955
